boa/src/builtins/array/mod.rs

```diff
@@ -399,6 +401,179 @@ impl Array {
+    pub(crate) fn from(
+        this: &JsValue,
+        args: &[JsValue],
+        context: &mut Context,
+    ) -> JsResult<JsValue> {
+        let items = args.get_or_undefined(0);
+        let mapfn = args.get_or_undefined(1);
+        let this_arg = args.get_or_undefined(2);
+
+        let mapping = if mapfn.is_undefined() {
+            // 2. If mapfn is undefined, let mapping be false
+            None
+        } else {
+            // 3. Else,
+            //     a. If IsCallable(mapfn) is false, throw a TypeError exception.
+            //     b. Let mapping be true.
+            Some(mapfn.as_callable().ok_or_else(|| {
+                context.construct_type_error(format!("{} is not a function", mapfn.display()))
+            })?)
+        };
+
+        if items.is_null_or_undefined() {
+            return context
+                .throw_type_error("Array.from requires an array-like object or iterator");
+        }
+
+        // 4. Let usingIterator be ? GetMethod(items, @@iterator).
+        let using_iterator = items
+            .get_method(WellKnownSymbols::iterator(), context)?
+            .map(JsValue::from);
+
+        if let Some(using_iterator) = using_iterator {
+            // 5. If usingIterator is not undefined, then
+
+            // a. If IsConstructor(C) is true, then
+            //     i. Let A be ? Construct(C).
+            // b. Else,
+            //     i. Let A be ? ArrayCreate(0en).
+            let a = match this.as_constructor() {
+                Some(constructor) => constructor
+                    .construct(&[], this, context)?
+                    .as_object()
+                    .cloned()
+                    .ok_or_else(|| {
+                        context.construct_type_error("Object constructor didn't return an object")
+                    })?,
+                _ => Self::array_create(0, None, context)?,
+            };
+
+            // c. Let iteratorRecord be ? GetIterator(items, sync, usingIterator).
+            let iterator_record =
+                items.get_iterator(context, Some(IteratorHint::Sync), Some(using_iterator))?;
+
+            // IfAbruptCloseIterator is a shorthand for a sequence of algorithm steps that use an Iterator Record
+            macro_rules! if_abrupt_close_iterator {
+                ($val:expr) => { match $val {
+                    // 2. Else if value is a Completion Record, set value to value.
+                    Ok(val) => val,
+                    // 1. If value is an abrupt completion, return ? IteratorClose(iteratorRecord, value).
+                    Err(err) => return iterator_record.close(Err(err), context)
+                } }
+            }
+
+            // d. Let k be 0.
+            // e. Repeat,
+            //     i. If k ≥ 2^53 - 1 (MAX_SAFE_INTEGER), then
+            //     ...
+            //     x. Set k to k + 1.
+            for k in 0..9_007_199_254_740_991_u64 {
```

Comment:
Let's use `boa::builtins::number::Number::MAX_SAFE_INTEGER` here, just to be safe ;)

Reply:
I wanted to use that but it's only there as a float and I didn't want to convert it to an integer. (Also the note about it being equal to `MAX_SAFE_INTEGER` is my own, not part of the spec, maybe I should make that clear 😅)